### PR TITLE
[SPARK-32080][SPARK-31998][SQL] Simplify ArrowColumnVector ListArray accessor

### PR DESCRIPTION
**Pull request by BryanCutler**
### What changes were proposed in this pull request?

This change simplifies the ArrowColumnVector ListArray accessor to use provided Arrow APIs available in v0.15.0 to calculate element indices.

### Why are the changes needed?

This simplifies the code by avoiding manual calculations on the Arrow offset buffer and makes use of more stable APIs.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing tests

### CODE REVIEW

**Comment by BryanCutler** [comment]
This also avoid having to deal with an API change in the ArrowBuf class for Arrow 1.0.0 and will allow Arrow-Spark integration tests to run again in preparation for the 1.0.0 release, see https://github.com/apache/arrow/pull/6316.

**Review by viirya — APPROVED** [review]
I think this also covers SPARK-31998?

**Review by HyukjinKwon — APPROVED** [review]
LGTM

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.0.

File: sql/catalyst/src/main/java/org/apache/spark/sql/vectorized/ArrowColumnVector.java [hunk 1]
```diff
@@ -458,10 +457,8 @@ final boolean isNullAt(int rowId) {
 
     @Override
     final ColumnarArray getArray(int rowId) {
-      ArrowBuf offsets = accessor.getOffsetBuffer();
-      int index = rowId * ListVector.OFFSET_WIDTH;
-      int start = offsets.getInt(index);
-      int end = offsets.getInt(index + ListVector.OFFSET_WIDTH);
+      int start = accessor.getElementStartIndex(rowId);
```
[the remark this answers is not in the document]
Comment (BryanCutler) [reply sql/catalyst/src/main/java/org/apache/spark/sql/vectorized/ArrowColumnVector.java:460]: Thanks @viirya , I wasn't aware SPARK-31998 was made, but yes that was the motivation for this fix.

**Comment by BryanCutler** [comment]
Thanks @HyukjinKwon and @viirya !

**Pull request closed without merging**